src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -209,8 +170,19 @@ private async Task ProcessAsynchronousWork()
 
                 // new work might be added before we check again as a result of waiting for all
                 // the child components to finish executing SetParametersAsync
-                await pendingWork;
-            };
+
+                try
+                {
+                    await pendingWork;
+                }
+                catch when (pendingWork.IsCanceled || HandleException(componentState.ComponentId, componentState.Component, pendingWork.Exception))
```

Comment:
Using `TaskCancelledException` doesn't always work. For instance, the exception thrown from cancelling a cancellation token results in the context of a Task results in the task transitioning to Canceled, but you still get the `OperationCancelledException`. Inspecting `IsCanceled` seems to be the reliable way to go